app/models/state_machines/assessment_attempt_state_machine.rb

```diff
@@ -12,7 +12,7 @@ class StateMachines::AssessmentAttemptStateMachine
   transition from: STATE_COMMENCED, to: STATE_PASSED
   transition from: STATE_COMMENCED, to: STATE_FAILED
 
-  after_transition(to: :failed) do |assessment_attempt, _transition|
+  after_transition(to: :passed) do |assessment_attempt, _transition|
```

Comment:
Just highlighting this for future reference; This might have been causing some of the confusion around the credly badges.

The code was firing when a level and CSA were failed rather than passed.